src/backend/distributed/planner/distributed_planner.c

```diff
@@ -157,6 +158,8 @@ distributed_planner(Query *parse,
 	bool fastPathRouterQuery = false;
 	Node *distributionKeyValue = NULL;
 
+	AttributeQueryIfAnnotated(query_string, parse->commandType);
```

Comment:
could we move it ot the end of this function? the logic here is quite important to understanding the planner and already quite messy and verbose